cmake/CompileOptions.cmake

```diff
@@ -151,5 +151,6 @@ if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" OR "${CMAKE_SYSTEM_NAME}" MATCHES "L
     set(DEFAULT_LINKER_OPTIONS
     PUBLIC
         -pthread
+        -ldl
```

Comment:
We should investigate if this is actually the required flag and for which platforms this has to be added.
It seems like my Ubuntu doesn't require this.